website/blog/2023-11-6-model-evaluation/index.md

```diff
@@ -0,0 +1,75 @@
+
+```python
+from typing import List 
+
+def below_zero(operations: List[int]) -> bool: 
+    
+    """ 
+    You're given a list of deposit and withdrawal operations on a bank account that starts with zero balance. Your task is to detect if at any point the balance of account fallls below zero, and at that point function should return True. Otherwise it should return False.
+    
+     >>> below_zero([1, 2, 3]) False 
+     
+     >>> below_zero([1, 2, -4, 5]) True 
+     
+    """
+```
+
+It was a pioneer research effort, but now suffers from some unfortunate drawbacks:
+
+1. ***Data is likely contaminated.*** HumanEval dataset has been around for over two years and it has been discussed and documented widely online. The latest coding LLMs are likely to have included its test data in training data crawling, which would make the evaluation no longer valid.
+
+2. ***Trivial coding questions that aren't mimicing real engineering setup.*** HumanEval inclues mostly LeetCode's interview-style questions, where they include a single function for LLMs to fill in the body. In a more realistic corporate setup, developers often add code in multiple files in a single PR, and constantly refer to functions implemented in other files. These are indeed more interesting yet challenging tasks for LLMs to perform, but are critical scenarios for AI coding assitants to land in enterprises.
+
+3. ***Limited coverage in programming languages.*** This one is obvious as HumanEval only includes Python code. We ❤️ all programming languages!
+
+Here are what we think a trustworthy evaluation setup should cover:
+
+1. ***Non-trivial code.*** Definitely no more Leetcode-style coding questions! The ideal evaluation should target projects with substantial engineering complexity. Evidences like lines of code, number of files, or number of contributors could serve as good indicators to estimate the code complexity.
+
+2. ***Cross-file references.*** This is a key factor to differentiate a more reliable and practical evaluation from something that only scratches the surface of the coding world. Engineers do not code in silo, but are greatly encouraged to reuse a function or API implemented by colleagues. (💡BTW, Tabby enabled [repository-level context support](https://tabby.tabbyml.com/blog/2023/10/16/repository-context-for-code-completion) in October, a feature exactly to get smarter coding suggestions in such scenario.)
```

Comment:
```suggestion
2. ***Cross-file references.*** This is a key factor to differentiate a more reliable and practical evaluation from something that only scratches the surface of the coding world. Engineers do not code in silo, but are greatly encouraged to reuse a function or API implemented by colleagues.
```

Let keep the story telling simple here (focusing on metric rather than efforts to improve the metric)